.github/workflows/build-test-deploy.yml

```diff
@@ -260,48 +262,85 @@ jobs:
+    - name: Login to DockerHub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
 
-      - name: Login to DockerHub
-        uses: docker/login-action@v1
-        with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
+    # the tests need the gprofiler image built (from Dockerfile). I run it separately here, because "docker build" prints the build logs
+    # more nicely. the tests will then be able to use the built image.
+    - name: Build gProfiler image
+      run: ./scripts/build_aarch64_container.sh -t gprofiler_aarch64 --output type=docker
 
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v1
+    - name: Export gProfiler image
+      run: mkdir -p output && docker image save gprofiler_aarch64 > output/gprofiler_aarch64.img
 
-      - name: Download executables from the previous job
-        uses: actions/download-artifact@v2
-        with:
-          name: gprofiler
-          path: output/
+    - name: Upload the image artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: gprofiler_aarch64.img
+        path: output/
+        retention-days: 1
 
-      # TODO: Add docker layer caching when GitHub Actions cache is stabilized and works good with "satackey/action-docker-layer-caching@v0.0.11"
-      - name: Build and push
-        run: |
-          set -x
+  test-container-aarch64:
+    needs: build-container-aarch64
 
-          BASE_IMAGE="${{ env.GH_REPO }}:${{ env.RELEASE_VERSION }}"
-          AARCH64_IMAGE="$BASE_IMAGE-aarch64"
+    runs-on: 
+      - self-hosted
+      - public
+      - ARM64
+    steps:
+  
+    - name: Login to DockerHub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
 
-          # build & push Aarch64
-          ./scripts/build_aarch64_container.sh --skip-exe-build --build-arg EXE_PATH=output/gprofiler_aarch64 -t "$AARCH64_IMAGE" --push
+    - name: Checkout Code
+      uses: actions/checkout@v3
+      with:
+        submodules: true
+
+    - name: Download the executable from previous job
+      uses: actions/download-artifact@v2
+      with:
+        name: gprofiler_aarch64.img
+        path: output/
 
+    - name: Import gProfiler image
+      run: docker image load < output/gprofiler_aarch64.img
+
+    - name: Extract resources from gProfiler executable
+      run: sudo docker run -v $PWD/gprofiler/resources:/app/gprofiler/resources gprofiler_aarch64 extract-resources --resources-dest=/app/gprofiler/resources
+
+    - name: setup runner requirements
+      run: ./scripts/setup_runner_requirements.sh
+
+    # TODO: Add docker layer caching when GitHub Actions cache is stabilized and works good with "satackey/action-docker-layer-caching@v0.0.11"
+    - name: Run gProfiler tests
```

Comment:
The x86_64 tests run on 3.8, 3.9, 3.10.

I'm actually planning to deprecate 3.8 and 3.9 soon so there's no need to test all 3 - let's focus on 3.10.

Can you use the GH action to install python? i.e
```
    steps:
      - name: Set up Python 3.10
        uses: actions/setup-python@v2
        with:
          python-version: "3.10"
```

Reply:
You know what, let's actually do it with `strategy` like done in the x86_64 tests:
```
    strategy:
      matrix:
        python-version:
          - "3.10"
```
this is extendable for a future where we'll want to add 3.11, and also shows up nicely in the job name (gonna be "test-container-aarch64 (3.10)" like we have for the x86 ones)